### PR TITLE
Edit functions for post analysis processing

### LINKED ISSUE #2: Create functions for post-analysis processing
Calculate offsetT and predictions and add them to the data frame (df). The processing of the residual_plot will be changed accordingly.

### PR DESCRIPTION
**Pull request by tomatine**
to deal with #2

### CODE REVIEW

File: sreftml/plots.py [hunk 1]
```diff
@@ -564,8 +558,8 @@ def residual_plot(
     n_biomarker = len(name_biomarkers)
     n_row, n_col = n2mfrow(n_biomarker, ncol_max)
     x_data = df.TIME.values + df.offsetT.values
-    y_pred = scaler_y.inverse_transform(sreft(scaled_features))
-    y_res = y_pred - df[name_biomarkers].values
+
+    y_res = df[[f"{biomarker}_pred" for biomarker in name_biomarkers]].values - df[name_biomarkers].values
```
Comment (RyotaJin) [inline sreftml/plots.py:570]: How about a warning prompting to use calculate_offsetT_prediction when offsetT or pred is not available?

File: sreftml/utilities.py [hunk 2]
```diff
@@ -343,3 +344,27 @@ def compute_permutation_importance(
...
+        name_biomarkers (list[str]): List of biomarker names.
+
+    Returns:
+        pd.DataFrame: The DataFrame with offsetT and prediction.
+    """
+    x_scaled, cov_scaled, m_scaled, y_scaled = scaled_features
+    offsetT = pd.DataFrame(sreft.model_1(np.concatenate((m_scaled, cov_scaled), axis=-1)),columns=["offsetT"])
+    y_pred = pd.DataFrame(scaler_y.inverse_transform(sreft(scaled_features)),
+                          columns=[f"{biomarker}_pred" for biomarker in name_biomarkers])
+    return pd.concat([offsetT, y_pred], axis=1)
```
Comment (RyotaJin) [inline sreftml/utilities.py]: I was assuming that the new columns would be added to the original data frame.

**Review by RyotaJin — CHANGES_REQUESTED** [review]

File: sreftml/plots.py (hunk 1 above)
Comment (tomatine) [reply sreftml/plots.py:570]: Good idea, I'll implement it.

File: sreftml/utilities.py (hunk 2 above)
Comment (tomatine) [reply sreftml/utilities.py]: That would be more convenient, I will do so.

**Review by RyotaJin — APPROVED** [review]
Confirmed. Thanks.

**Pull request merged**